### PR TITLE
[SOAR-17883] rapid7_insightcloudsec, SDK bump, adding a new functionality named get resource id

### PR DESCRIPTION
**Pull request by lcwiklinski-r7**
### Description
https://rapid7.atlassian.net/browse/SOAR-17883

Describe the proposed changes:

  - Updated SDK to the latest version
  - Adding a new functionality named `get resource id`

#### Unit Tests
I tested unit tests / added tests to cover `get_resource_id` locally and they all passed.
image

### CODE REVIEW

File: plugins/rapid7_insightcloudsec/help.md [hunk 1]
```diff
@@ -421,12 +494,16 @@ _This plugin does not contain any troubleshooting information._
...
 # Links
 
+* [Rapid7 InsightCloudSec](https://github.com/rapid7/insightconnect-plugins/tree/master/plugins/rapid7_insightcloudsec)
+* [Docs](https://docs.rapid7.com/insightcloudsec/api/v2/)
+
 ## References
 
 * [Rapid7 InsightCloudSec](https://www.rapid7.com/products/insightcloudsec/)
-
+* You can find testing credentials under the Shared-Integration Lab/DivvyCloud - Demo @ 1Password
```
Comment (igorski-r7) [inline plugins/rapid7_insightcloudsec/help.md:509]: Imo this should be removed

File: plugins/rapid7_insightcloudsec/unit_test/test_get_resource_id.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+
+@patch("requests.request", side_effect=Util.mocked_requests)
+class TestGetResourceId(TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.action = Util.default_connector(GetResourceId())
+
+    @parameterized.expand(Util.load_parameters("get_resource_id").get("parameters"))
+    def test_get_resource_id(self, mock_request, name, limit, offset, search_string, expected):
```
Comment (igorski-r7) [inline plugins/rapid7_insightcloudsec/unit_test/test_get_resource_id.py:24]: Could you please add type-hints here? 😄

File: plugins/rapid7_insightcloudsec/unit_test/test_get_resource_id.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        actual = self.action.run({Input.LIMIT: limit, Input.OFFSET: offset, Input.SEARCH_STRING: search_string})
+        validate(actual, self.action.output.schema)
+        self.assertEqual(actual, expected)
+
+    @parameterized.expand(Util.load_parameters("get_resource_id_bad").get("parameters"))
+    def test_get_resource_id_bad(self, mock_request, name, limit, offset, search_string, cause, assistance):
```
Comment (igorski-r7) [inline plugins/rapid7_insightcloudsec/unit_test/test_get_resource_id.py:30]: Same here

File: plugins/rapid7_insightcloudsec/unit_test/responses/get_resource_id.json.resp
Comment (igorski-r7) [inline plugins/rapid7_insightcloudsec/unit_test/responses/get_resource_id.json.resp:1]: Looks good. Could you please add new-line at the end of file?

File: plugins/rapid7_insightcloudsec/unit_test/parameters/get_resource_id_bad.json.resp
Comment (igorski-r7) [inline plugins/rapid7_insightcloudsec/unit_test/parameters/get_resource_id_bad.json.resp:1]: Same here -> New line at the end of file

File: plugins/rapid7_insightcloudsec/unit_test/parameters/get_resource_id.json.resp
Comment (igorski-r7) [inline plugins/rapid7_insightcloudsec/unit_test/parameters/get_resource_id.json.resp:1]: Same here

File: plugins/rapid7_insightcloudsec/plugin.spec.yaml
Comment (igorski-r7) [inline plugins/rapid7_insightcloudsec/plugin.spec.yaml:1]: Here as well, could you please add new-line at the end of file?

File: plugins/rapid7_insightcloudsec/help.md (hunk 1 above)
Comment (lcwiklinski-r7) [reply plugins/rapid7_insightcloudsec/help.md:509]: @dlaverty-r7 asked me to add information about where the Api-Key comes from for testing, as the name DivvyCloud may not be obvious. Dympna, can you confirm if it's okay?

Comment (dlaverty-r7) [reply plugins/rapid7_insightcloudsec/help.md:509]: Ah sorry, @lcwiklinski-r7 I should have clarified, I meant can you add a comment in our 1password creds store? This comment can be removed from the help.md as all of this code is visible by customers.

File: plugins/rapid7_insightcloudsec/unit_test/test_get_resource_id.py (hunk 3 above)
Comment (lcwiklinski-r7) [reply plugins/rapid7_insightcloudsec/unit_test/test_get_resource_id.py]: good point 👍 done

File: plugins/rapid7_insightcloudsec/unit_test/test_get_resource_id.py (hunk 2 above)
Comment (lcwiklinski-r7) [reply plugins/rapid7_insightcloudsec/unit_test/test_get_resource_id.py]: good point 👍 done

File: plugins/rapid7_insightcloudsec/unit_test/parameters/get_resource_id.json.resp
Comment (lcwiklinski-r7) [reply plugins/rapid7_insightcloudsec/unit_test/parameters/get_resource_id.json.resp:1]: good point 👍 done

File: plugins/rapid7_insightcloudsec/plugin.spec.yaml
Comment (lcwiklinski-r7) [reply plugins/rapid7_insightcloudsec/plugin.spec.yaml:1]: good point 👍 done

File: plugins/rapid7_insightcloudsec/unit_test/responses/get_resource_id.json.resp
Comment (lcwiklinski-r7) [reply plugins/rapid7_insightcloudsec/unit_test/responses/get_resource_id.json.resp:1]: good point 👍 done

File: plugins/rapid7_insightcloudsec/help.md (hunk 1 above)
Comment (lcwiklinski-r7) [reply plugins/rapid7_insightcloudsec/help.md]: ok, done 👍

**Review by igorski-r7 — APPROVED** [review]

**Pull request merged**